modules/afprog/afprog.c

```diff
@@ -526,6 +526,7 @@ afprogram_dd_restore_reload_store_item(AFProgramDestDriver *self, GlobalConfig *
     {
       self->process_info.pid = restored_info->pid;
       self->writer = restored_info->writer;
+      ((LogPipe *) self->writer)->cfg = cfg;
```

Comment:
We could call `log_pipe_set_config()` in `init()`, right before `log_writer_set_options()`, to make sure we use the appropriate config instance no matter which code path we are coming from (this is what we do in other modules).